native/cocos/core/animation/SkeletalAnimationUtils.h

```diff
@@ -25,18 +25,21 @@
 
 #pragma once
 
-#include "math/Mat4.h"
+#include <utility>
 
+#include "math/Mat4.h"
+#include "cocos/base/RefCounted.h"
+#include "cocos/base/Ptr.h"
 namespace cc {
 
 class Node;
 
-struct IJointTransform {
-    Node *node{nullptr};
+struct IJointTransform : public RefCounted {
+    Node* node{nullptr};
     Mat4 local;
     Mat4 world;
     int stamp{-1};
-    struct IJointTransform *parent{nullptr};
+    IntrusivePtr<IJointTransform> parent;
```

Comment:
Will parent also hold a reference of its child?

Reply:
@dumganhar I'm not sure, should use IntrusivePtr for parent or just raw ptr?

Reply:
The ownership here is complicated, we need to check the original ts code.

Reply:
@mmyduckx Please separate this PR into two.
The `imageAssetProto.destroy` one could be merged first.